examples/next-openai-pages/pages/chat/stream-chat-attachments/index.tsx

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable @next/next/no-img-element */
+import { useChat } from 'ai/react';
+import { useRef, useState } from 'react';
+
+export default function Page() {
+  const { messages, input, handleSubmit, handleInputChange, isLoading } =
+    useChat({
+      api: '/api/stream-chat',
+    });
+
+  const [files, setFiles] = useState<FileList | undefined>(undefined);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex flex-col p-2 gap-2">
+        {messages.map(message => (
+          <div key={message.id} className="flex flex-row gap-2">
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+
+            <div className="flex flex-col gap-2">
+              {message.content}
+
+              <div className="flex flex-row gap-2">
+                {message.experimental_attachments?.map((attachment, index) =>
+                  attachment.contentType?.includes('image/') ? (
+                    <img
+                      key={`${message.id}-${index}`}
+                      className="w-24 rounded-md"
+                      src={attachment.url}
+                      alt={attachment.name}
+                    />
+                  ) : attachment.contentType?.includes('text/') ? (
+                    <div className="w-32 h-24 rounded-md text-xs ellipsis overflow-hidden p-2 text-zinc-500 border">
+                      {Buffer.from(
+                        attachment.url.split(',')[1],
+                        'base64',
+                      ).toString('utf-8')}
```

Comment:
does this work in browers?

Reply:
added a helper function [here](https://github.com/vercel/ai/pull/2226/files#diff-02c72a102d4cbb0de457bbb8233925ac1640bf37d02e6be326dd0990eb6bf1af).

it uses [window.atob](https://developer.mozilla.org/en-US/docs/Web/API/Window/atob) instead, which works in browsers!

only `atob` of node.js is deprecated ([source](https://stackoverflow.com/a/70851350))